lib/optimizely/config_manager/http_project_config_manager.rb

```diff
@@ -0,0 +1,223 @@
+
+    attr_reader :config
+
+    # Initialize config manager. One of sdk_key or url has to be set to be able to use.
+    #
+    # sdk_key - Optional string uniquely identifying the datafile. It's required unless a URL is passed in.
+    # datafile: Optional JSON string representing the project.
+    # polling_interval - Optional floating point number representing time interval in seconds
+    #                  at which to request datafile and set ProjectConfig.
+    # blocking_timeout -
+    # auto_update -
+    # start_by_default -
+    # url - Optional string representing URL from where to fetch the datafile. If set it supersedes the sdk_key.
+    # url_template - Optional string template which in conjunction with sdk_key
+    #               determines URL from where to fetch the datafile.
+    # logger - Provides a logger instance.
+    # error_handler - Provides a handle_error method to handle exceptions.
+    # skip_json_validation - Optional boolean param which allows skipping JSON schema
+    #                       validation upon object invocation. By default JSON schema validation will be performed.
+    def initialize(
+      sdk_key: nil,
+      url: nil,
+      url_template: nil,
+      polling_interval: nil,
+      blocking_timeout: nil,
+      auto_update: true,
+      start_by_default: true,
+      datafile: nil,
+      logger: nil,
+      error_handler: nil,
+      skip_json_validation: false,
+      notification_center: nil
+    )
+      @logger = logger || NoOpLogger.new
+      @error_handler = error_handler || NoOpErrorHandler.new
+      @datafile_url = get_datafile_url(sdk_key, url, url_template)
+      @polling_interval = nil
+      polling_interval(polling_interval)
+      blocking_timeout ||= Helpers::Constants::CONFIG_MANAGER['BLOCKING_TIMEOUT']
+      @blocking_timeout = blocking_timeout
+      @last_modified = nil
+      @async_scheduler = AsyncScheduler.new(method(:fetch_datafile), @polling_interval, auto_update, @logger)
+      @async_scheduler.start! if start_by_default == true
+      @skip_json_validation = skip_json_validation
+      @notification_center = notification_center.is_a?(Optimizely::NotificationCenter) ? notification_center : NotificationCenter.new(@logger, @error_handler)
+      @config = datafile.nil? ? nil : DatafileProjectConfig.create_project_config_from_datafile(datafile, @logger, @error_handler, @skip_json_validation)
+      @mutex = Mutex.new
+      @resource = ConditionVariable.new
+    end
+
+    def ready?
+      !@config.nil?
+    end
+
+    def start!
+      @async_scheduler.start!
+    end
+
+    def stop!
+      @async_scheduler.stop!
+    end
+
+    def get_config
+      # Get Project Config.
+
+      # Returns config immediately if the config has been initialized. When a hardcoded
+      # datafile is passed on init, config becomes ready immediately.
+
+      return @config if ready?
```

Comment:
This shouldn't directly return config. Please check this condition in C# or java. 
If manager is not started and get_config is called, it should always poll regardless config was already set or nil.

Reply:
https://github.com/optimizely/csharp-sdk/blob/master/OptimizelySDK/Config/PollingProjectConfigManager.cs#L108